autogpt/commands/image_gen.py

```diff
@@ -60,20 +62,40 @@ def generate_image_with_hf(prompt: str, filename: str) -> str:
         "X-Use-Cache": "false",
     }
 
-    response = requests.post(
-        API_URL,
-        headers=headers,
-        json={
-            "inputs": prompt,
-        },
-    )
-
-    image = Image.open(io.BytesIO(response.content))
-    print(f"Image Generated for prompt:{prompt}")
-
-    image.save(filename)
+    retry_count = 0
+    while retry_count < 10:
+        response = requests.post(
+            API_URL,
+            headers=headers,
+            json={"inputs": prompt},
+        )
 
-    return f"Saved to disk:{filename}"
+        if response.ok:
+            try:
+                image = Image.open(io.BytesIO(response.content))
+                print(f"Image Generated for prompt:{prompt}")
+                image.save(filename)
+                return f"Saved to disk:{filename}"
+            except Exception as e:
+                print(e)
+                break
+        else:
+            try:
+                error = json.loads(response.text)
+                if "estimated_time" in error:
+                    delay = error["estimated_time"]
+                    print(response.text)
+                    print("Retrying in", delay)
+                    time.sleep(delay)
```

Comment:
this is gonna trip up someone when async rolls out :(

Reply:
What if we make this configurable via a .env setting, with a default to enabled?